include/nighthawk/client/process_context.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "envoy/network/address.h"
+#include "envoy/stats/store.h"
+
+#include "nighthawk/client/client_worker.h"
+#include "nighthawk/client/factories.h"
+#include "nighthawk/client/options.h"
+#include "nighthawk/client/output_formatter.h"
+#include "nighthawk/common/statistic.h"
+#include "nighthawk/common/uri.h"
+
+#include "common/api/api_impl.h"
+#include "common/common/logger.h"
+#include "common/event/real_time_system.h"
+
+namespace Nighthawk {
+namespace Client {
+
+/**
+ * Process context is shared between the CLI and grpc service. It is capable of executing
+ * a full Nighthawk test run.
+ */
+class ProcessContext {
+public:
+  virtual ~ProcessContext() = default;
+
+  /**
+   * @return uint32_t the number of workers that will be used based on configuration
+   * and available hardware.
+   */
+  virtual uint32_t determineConcurrency() const PURE;
+
+  /**
+   * @return Envoy::Event::TimeSystem& The time system implementation used by this instance.
+   */
+  virtual Envoy::Event::TimeSystem& time_system() PURE;
+
+  /**
+   * Envoy::Api::Impl& The api implementation used by this instance.
+   */
+  virtual Envoy::Api::Impl& api() PURE;
```

Comment:
Why do we export these out of the `ProcessContext`? AFAICT, `ProcessContext` main goal is to encapsulate `run` and perform a full run of a workload. Why do we need to have these other features in the interface?

Reply:
You're right. Cleaned this up, and purged them from the interface.